### PR TITLE
(FLW-1603) account create option for import unique chain

### PR DESCRIPTION
**Pull request by Viscount88**
account create option for import unique chain

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -230,4 +230,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: b797fc981d66949081e91425d8a47bd8d714d965
 
-COCOAPODS: 1.11.3
+COCOAPODS: 1.11.2
```
Comment (DRadmir) [inline Podfile.lock:233]: update cocoa pods, please

File: fearless/Common/Model/ReplaceChainOption.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+import UIKit
+
+enum ReplaceChainOption: CaseIterable {
+    case create
+    case `import`
+
+    var icon: UIImage? {
+        R.image.iconReplace()
```
Comment (DRadmir) [inline fearless/Common/Model/ReplaceChainOption.swift:9]: this is right image?

File: fearless/Modules/AccountCreate/AccountCreatePresenter.swift [hunk 3]
```diff
@@ -128,10 +135,16 @@ final class AccountCreatePresenter {
 extension AccountCreatePresenter: AccountCreatePresenterProtocol {
     func setup() {
         interactor.setup()
-
+        switch flow {
+        case .wallet:
+            view?.set(chainType: .both)
+        case let .chain(model):
+            selectedCryptoType = CryptoType(rawValue: UInt8(model.meta.substrateCryptoType)) ?? .sr25519
```
Comment (bnsports) [inline fearless/Modules/AccountCreate/AccountCreatePresenter.swift:142]: how many "?? .sr25519" we have? maybe it makes sense to create CryptoType.DefaultSubstrate constants?

File: fearless/Modules/AccountCreate/AccountCreateViewFactory.swift [hunk 4]
```diff
@@ -32,19 +35,22 @@ final class AccountCreateViewFactory: AccountCreateViewFactoryProtocol {
...
         return createViewForUsername(
             model: model,
+            flow: .wallet,
             wireframe: wireframe
         )
     }
 
     static func createViewForUsername(
         model: UsernameSetupModel,
+        flow: AccountCreateFlow = .wallet,
```
Comment (bnsports) [inline fearless/Modules/AccountCreate/AccountCreateViewFactory.swift:45]: is default value needs there ?

File: fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift [hunk 5]
```diff
@@ -2,7 +2,11 @@ import Foundation
 
 extension AddAccount {
     final class UsernameSetupWireframe: UsernameSetupWireframeProtocol {
-        func proceed(from view: UsernameSetupViewProtocol?, model: UsernameSetupModel) {
+        func proceed(
+            from view: UsernameSetupViewProtocol?,
+            flow _: AccountCreateFlow = .wallet,
```
Comment (bnsports) [inline fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift:7]: looks like flow parameter unused ?

File: fearless/Modules/SwitchAccount/SwitchAccount+UsernameSetupWireframe.swift [hunk 6]
```diff
@@ -2,7 +2,11 @@ import Foundation
 
 extension SwitchAccount {
     final class UsernameSetupWireframe: UsernameSetupWireframeProtocol {
-        func proceed(from view: UsernameSetupViewProtocol?, model: UsernameSetupModel) {
+        func proceed(
+            from view: UsernameSetupViewProtocol?,
+            flow _: AccountCreateFlow = .wallet,
```
Comment (bnsports) [inline fearless/Modules/SwitchAccount/SwitchAccount+UsernameSetupWireframe.swift:7]: flow parameter unused ?

File: fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift (hunk 5 above)
Comment (DRadmir) [reply fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift:7]: @bnsports this is protocol method, sorry

File: fearless/Modules/AccountCreate/AccountCreatePresenter.swift (hunk 3 above)
Comment (Viscount88) [reply fearless/Modules/AccountCreate/AccountCreatePresenter.swift:142]: 5. May be

File: fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift (hunk 5 above)
Comment (Viscount88) [reply fearless/Modules/AddAccount/Wireframes/AddAccount+UsernameSetupWireframe.swift:7]: yep. we don't use it here, but protocols require it

File: fearless/Modules/SwitchAccount/SwitchAccount+UsernameSetupWireframe.swift (hunk 6 above)
Comment (Viscount88) [reply fearless/Modules/SwitchAccount/SwitchAccount+UsernameSetupWireframe.swift:7]: same thing

File: fearless/Common/Model/ReplaceChainOption.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+enum ReplaceChainOption: CaseIterable {
+    case create
+    case `import`
+
+    var icon: UIImage? {
+        switch self {
+        case .create:
+            return R.image.iconAdd()
+        case .import:
+            return R.image.iconReplace()
```
Comment (bnsports) [inline fearless/Common/Model/ReplaceChainOption.swift:13]: is it correct icon ?

**Review by bnsports — APPROVED** [review]

**Pull request merged**